DOCS-12470 Salesforce CDP 1.1 (#1685)
mulesoft/docs-connectors: salesforce-cdp/1.1/modules/ROOT/pages/salesforce-cdp-connector-examples.adoc

Shows how to retrieve current status of the specified job ID.
** <<Delete Job - Bulk API>> +
Shows how to delete or close the specified job ID, which deletes job data and metadata stored by Salesforce.
* Query API Flows include: +
** <<Query API>> +
Shows how to query the Salesforce CDP data lake across data model, lake, unified, and linked objects.
* Calculated Insights API Flows include: +
** <<Insights - List Metadata API>> +
Shows how to retrieve the metadata including the dimensions and measures for all calculated insights.
** <<Insights - Get Metadata API>> +
Shows how to retrieve the metadata including the dimensions and measures for a calculated insight.
** <<Insights - Get Insights API>> +
Shows how to query the calculated insights. Users can slice, dice, and filter by selecting different dimensions, measures, and filters.
* Profile API Flows include: +
** <<Profile - List Metadata API>> +
Shows how to retrieve the metadata for all data model objects.
** <<Profile - Get Metadata API>> +
Shows how to retrieve the metadata for the requested data model object.
** <<Profile - Search Records API>> +
Shows how to retrieve data model object records based on search filters.

> ** <<Profile - Search Records By Id API>> +

Reviewer: Looks like this should be: Profile - Search Records By ID API. The link didn't work in my IntelliJ adoc previewer.

Shows how to retrieve data model object records based on indexes and search filters.
** <<Profile - Search Records With Child Records API>> +
Shows how to retrieve data model object records and child object records based on indexes and search filters.
** <<Profile - Search Records With Insight API>> +
Shows how to retrieve data model object records and a computed view based on indexes and search filters.

> ** <<Data Action Webhook>> +

Reviewer: Data Action Webhook isn't really grouped with the "Profile API Flows". I'd suggest keeping it by itself without a sublist or creating a sublist just for it with something like "Data Action API Flows".

** <<Profile - Search Records With Child Records API>> +
Shows how to retrieve data model object records and child object records based on indexes and search filters.
** <<Profile - Search Records With Insight API>> +
Shows how to retrieve data model object records and a computed view based on indexes and search filters.
* Data Action API Flows include: +
** <<Data Action Webhook>> +
Shows how to run operations when a data action is triggered.

== Insert Data - Streaming API

This Mule flow shows how to insert data via the streaming API to an object's endpoint. Refer to the https://developer.salesforce.com/docs/atlas.en-us.c360a_api.meta/c360a_api/c360a_api_insert_records.htm[Salesforce API docs].

This example uses the following operations:

* *HTTP Listener* +
Accepts data from HTTP POST requests with the source and object name as the URI parameters.
* *Logger* +
Shows the HTTP request received from the *Listener*, then later shows the HTTP response from the *Insert Objects* operation.
* *Streaming Insert*
+
** Authenticates based on your choice of either JWT or username and password.
** Receives the source name, object name, and payload.

> ** Calls the Salesforce API to insert the payload to that object's endpoint.

Reviewer 2: This wording isn't quite right, but it's up to you whether you want to change it here and elsewhere. The meaning is clear.

Suggestion: Requests that the Salesforce API insert...

	<sdc:sdc-config name="Salesforce_CDP_OAuth_UsernamePassword_config" doc:name="Salesforce CDP config" >
		<sdc:oauth-user-pass-connection clientId="${server.consumerKey}" clientSecret="${server.consumerSecret}" username="${server.userName}" password="${server.password}" audienceUrl="${server.audienceUrl}" />
	</sdc:sdc-config>
	<flow name="DeleteJob" >
		<http:listener doc:name="Delete /jobs/delete/{jobId}" config-ref="HTTP_Listener_config" path="/jobs/delete/{jobId}"/>
		<sdc:delete-bulk-job doc:name="Delete Job" config-ref="Salesforce_CDP_OAuth_JWT_config" idUriParam="#[attributes.uriParams.jobId]"/>
		<logger level="INFO" doc:name="Logger" />
	</flow>
</mule>

----
=== Steps for Running This Example
. https://help.salesforce.com/s/articleView?id=sf.c360_a_connect_an_ingestion_source.htm&type=5[Verify that your connector is configured.]
. Save the project.
. <<Create Job - Bulk API, Create a job>> and copy the resulting job ID.
. Use the job you copied to close the job (see <<Close Job - Bulk API>> flow).
. Test the flow by sending a `DELETE` to
`localhost:8081/jobs/delete/{JOB_ID}`, using the job ID that you copied earlier.

== Insights - List Metadata API

> This Mule flow shows how to get the metadata including the dimensions and filters for all calculated insights.

Reviewer 2: ```suggestion
This Mule flow shows how to get the metadata, including the dimensions and filters for all calculated insights.
```

> This Mule flow shows how to get the metadata including the dimensions and filters for all calculated insights.

Reviewer 2: There are other occurrences of the text above  text to look for.

This example uses the following operations:

* *HTTP Listener* +
Accepts HTTP GET requests.
* *Insights - List Metadata*
+

> ** Authenticates based on your choice of either JWT or username and password.

Reviewer 2: ```suggestion
** Authenticates based on your choice of either a JWT or a username and password.
```

Reviewer 2: There are other occurrences of the above text.

	</sdc:sdc-config>
	<configuration-properties doc:name="Configuration properties" file="mule-app.properties" />
	<sdc:sdc-config name="Salesforce_CDP_OAuth_UsernamePassword_config" doc:name="Salesforce CDP config" >
		<sdc:oauth-user-pass-connection clientId="${server.consumerKey}" clientSecret="${server.consumerSecret}" username="${server.userName}" password="${server.password}" audienceUrl="${server.audienceUrl}" />
	</sdc:sdc-config>
	<flow name="List_Calculated_Insights_Metadata" >
		<http:listener doc:name="GET /insight/metadata" allowedMethods="GET" path="/insight/metadata" config-ref="HTTP_Listener_config"/>
		<sdc:get-all-calculated-insight-metadata doc:name="Insights - List Metadata"  config-ref="Salesforce_CDP_OAuth_JWT_config"/>
	</flow>
</mule>
----

=== Steps for Running This Example

. Verify that your connector is configured.
. Save the project.
. Test the flow by sending a `GET` to
`localhost:8081/insight/metadata`.

== Insights - Get Metadata API

> This Mule flow shows how to retrieve the metadata including the dimensions and filters for a calculated insight. Refer to the https://developer.salesforce.com/docs/atlas.en-us.c360a_api.meta/c360a_api/c360a_api_insights_meta_ci_name.htm[Salesforce CDP API Developer Guide] for usage and further details.

Reviewer 2: ```suggestion
This Mule flow shows how to retrieve the metadata, including the dimensions and filters for a calculated insight. Refer to the https://developer.salesforce.com/docs/atlas.en-us.c360a_api.meta/c360a_api/c360a_api_insights_meta_ci_name.htm[Salesforce CDP API Developer Guide] for usage and further details.
```

	</sdc:sdc-config>
	<flow name="Get_Calculated_Insights" >
		<http:listener doc:name="GET /insight/calculated-insights/{ci_name}" config-ref="HTTP_Listener_config" path="/insight/calculated-insights/{ci_name}" allowedMethods="GET"/>
		<sdc:get-calculated-insight-with-filters-fields-and-limit doc:name="Insights - Get Insights" config-ref="Salesforce_CDP_OAuth_JWT_config" ciNameUriParam="#[attributes.uriParams.ci_name]" dimensionsQueryParam="#[attributes.queryParams.dimensions]" measuresQueryParam="#[attributes.queryParams.measures]" limitQueryParam="#[attributes.queryParams.limit]" offsetQueryParam="#[attributes.queryParams.offset]" filtersQueryParam="#[attributes.queryParams.filters]" orderbyQueryParam="#[attributes.queryParams.orderby]" timeGranularityQueryParam="#[attributes.queryParams.timeGranularity]"/>
	</flow>
</mule>
----

=== Steps for Running This Example

. Verify that your connector is configured.
. Save the project.
. Test the flow by sending a `GET` to
`localhost:8081/insight/calculated-insights/{CALCULATED_INSIGHT_NAME}`.

== Profile - List Metadata API

> This Mule flow shows how to retrieve the list of data model objects, their fields, and category. Refer to the https://developer.salesforce.com/docs/atlas.en-us.c360a_api.meta/c360a_api/c360a_api_profile_meta.htm[Salesforce CDP API Developer Guide] for usage and further details.

Reviewer 2: ```suggestion
This Mule flow shows how to retrieve the list of data model objects and their fields and category. Refer to the https://developer.salesforce.com/docs/atlas.en-us.c360a_api.meta/c360a_api/c360a_api_profile_meta.htm[Salesforce CDP API Developer Guide] for usage and further details.
```

Reviewer 2: Ditto with other occurrences of this text.

		<sdc:get-computed-view-for-profile doc:name="Profile - Search Records With Insight" config-ref="Salesforce_CDP_OAuth_JWT_config" dataModelNameUriParam="#[attributes.uriParams.dataModelName]" idUriParam="#[attributes.uriParams.id]" ciNameUriParam="#[attributes.uriParams.ci_name]" searchKeyQueryParam="#[attributes.queryParams.searchKey]" dimensionsQueryParam="#[attributes.queryParams.dimensions]" measuresQueryParam="#[attributes.queryParams.measures]" limitQueryParam="#[attributes.queryParams.limit]" filtersQueryParam="#[attributes.queryParams.filters]" offsetQueryParam="#[attributes.queryParams.offset]" orderbyQueryParam="#[attributes.queryParams.orderby]" timeGranularityQueryParam="#[attributes.queryParams.timeGranularity]"/>
	</flow>
</mule>
----

=== Steps for Running This Example

. Verify that your connector is configured.
. Save the project.
. Test the flow by sending a `GET` to
`localhost:8081/profile/{DATA_MODEL_NAME}/{Id}/calculated-insights/{CALCULATED_INSIGHT_NAME}`.

== Data Action Webhook

> This Mule flow shows how to create a webhook that will log its payload when triggered by a Data Action. Refer to the https://help.salesforce.com/s/articleView?language=en_US&type=5&id=sf.c360_a_data_actions.htm[Salesforce Data Action docs].

Reviewer 2: ```suggestion
This Mule flow shows how to create a webhook that logs its payload when triggered by a Data Action. Refer to the https://help.salesforce.com/s/articleView?language=en_US&type=5&id=sf.c360_a_data_actions.htm[Salesforce Data Action docs].
```

	xmlns:doc="http://www.mulesoft.org/schema/mule/documentation"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="
    http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
    http://www.mulesoft.org/schema/mule/sdc http://www.mulesoft.org/schema/mule/sdc/current/mule-sdc.xsd">
	<flow name="Data_Action_Webhook" doc:id="3072197f-ee1f-4ed5-99d2-a9fa62230dbf" >
		<sdc:webhook-listener doc:name="Data Action Webhook" doc:id="fe3a924a-ba94-447d-9d3c-df079fc01de2" config-ref="Salesforce_CDP_Sdc_webhook_config" path="${webhook.path}" signingKey="${webhook.signingKey}" signingAlgorithm="HmacSHA256"/>
		<logger level="INFO" doc:name="Log Payload" doc:id="716538d3-7d06-4e9c-a4f7-b0d350b4dafe" message="#[output application/json --- payload]"/>
	</flow>
</mule>
----

=== Steps for Running This Example

. Verify you have https://help.salesforce.com/s/articleView?id=sf.c360_a_create_streaming_insight.htm&type=5[created a Streaming Insight].
. Verify you https://help.salesforce.com/s/articleView?id=sf.c360_a_data_actions.htm&type=5[created a Data Action Target] with an Action Target Type of `Webhook`.

> . Verify you https://help.salesforce.com/s/articleView?id=sf.c360_a_data_actions.htm&type=5[created a Data Action] running on a condition against your streaming Calculated Insight, that will use the Data Action Target.

Reviewer 2: This text is confusing. Maybe:
Verify that you https://help.salesforce.com/s/articleView?id=sf.c360_a_data_actions.htm&type=5[created a Data Action that uses the Data Action Target and that runs on a condition against your streaming Calculated Insight.

(I am not sure if my rewrite makes sense in the context of the product.) What does it mean to "run on a condition"?
